client/app/components/QueryOwnerEditorDialog/index.jsx

```diff
@@ -0,0 +1,147 @@
+import { toHuman } from "@/lib/utils";
+import HelpTrigger from "@/components/HelpTrigger";
+import { UserPreviewCard } from "@/components/PreviewCard";
+import PlainButton from "@/components/PlainButton";
+import notification from "@/services/notification";
+import User from "@/services/user";
+
+import "./index.less";
+
+const { Option } = Select;
+const DEBOUNCE_SEARCH_DURATION = 200;
+
+const searchUsers = searchTerm =>
+  User.query({ q: searchTerm })
+    .then(({ results }) => results)
+    .catch(() => []);
+
+function OwnerEditorDialogHeader({ context }) {
+  return (
+    <>
+      Update Query Owner
+      <div className="modal-header-desc">
+        {`Updating the ${context} owner is enabled for the author of the query and for admins. `}
+      </div>
+    </>
+  );
+}
+
+OwnerEditorDialogHeader.propTypes = { context: PropTypes.oneOf(["query"]) };
+OwnerEditorDialogHeader.defaultProps = { context: "query" };
+
+function UserSelect({ onSelect, shouldShowUser }) {
+  const [loadingUsers, setLoadingUsers] = useState(true);
+  const [users, setUsers] = useState([]);
+  const [searchTerm, setSearchTerm] = useState("");
+
+  const debouncedSearchUsers = useCallback(
+    debounce(
+      search =>
+        searchUsers(search)
+          .then(setUsers)
+          .finally(() => setLoadingUsers(false)),
+      DEBOUNCE_SEARCH_DURATION
+    ),
+    []
+  );
+
+  useEffect(() => {
+    setLoadingUsers(true);
+    debouncedSearchUsers(searchTerm);
+  }, [debouncedSearchUsers, searchTerm]);
+
+  return (
+    <Select
+      className="w-100 m-b-10"
+      placeholder="Select new owner..."
+      showSearch
+      onSearch={setSearchTerm}
+      suffixIcon={
+        loadingUsers ? (
+          <span role="status" aria-live="polite" aria-relevant="additions removals">
+            <i className="fa fa-spinner fa-pulse" aria-hidden="true" />
+            <span className="sr-only">Loading...</span>
+          </span>
+        ) : (
+          <i className="fa fa-search" aria-hidden="true" />
+        )
+      }
+      filterOption={false}
+      notFoundContent={null}
+      value={undefined}
+      getPopupContainer={trigger => trigger.parentNode}
+      onSelect={onSelect}>
+      {users.filter(shouldShowUser).map(user => (
+        <Option key={user.id} value={user.id}>
+          <UserPreviewCard user={user} />
+        </Option>
+      ))}
+    </Select>
+  );
+}
+
+UserSelect.propTypes = {
+  onSelect: PropTypes.func,
+  shouldShowUser: PropTypes.func,
+};
+UserSelect.defaultProps = { onSelect: () => {}, shouldShowUser: () => true };
+
+function OwnerEditorDialog({ dialog, author, context }) {
+
+  const [owner, setOwner] = useState(author);
+
+  const loadOwner = useCallback((userId) => {
+    User.get({ id: userId }).then(user => setOwner(user));
+  }, []);
+
+  const userIsOwner = useCallback(
+    user => user.id === author.id,
+    [author.id]
+  );
+
+  // useEffect(() => {
```

Comment:
Should this be in here? :smile:

Reply:
Oh good catch, I commented this out when working on this as I wasn't sure what exactly it was there for. I modeled this index file after the PermissionsEditorDialog index file, in which the loadOwner function is akin to loadUsersWithPermission function. It used the same useEffect statement, so I could add this back. 

Do you know what this code block was necessary for the PermissionsEditorDialog component by chance?

Reply:
> Do you know why this code block was necessary for the PermissionsEditorDialog component by chance?

Unfortunately no.  At least "not yet" as I still need to get around to learning React properly.  Am more of a backend guy than frontend dev at this stage. :wink:

Someone else might know though. :smile: